layers/sip.go

```diff
@@ -245,6 +245,9 @@ func (s *SIP) NextLayerType() gopacket.LayerType {
 
 // DecodeFromBytes decodes the slice into the SIP struct.
 func (s *SIP) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+	// There are no further layers, so set BaseLayer's Contents should point at
+	// this layer.
+	s.BaseLayer = BaseLayer{Contents: data[:len(data)]}
```

Comment:
Well I agree, that `Contents` is missing, but "no further leayers" is not completely true (e.g., INVITE or 200 OK with SDP attached):
https://github.com/google/gopacket/blob/90781af512195c8b22c46db7cce068bc30ef8394/layers/sip.go#L280-L283

Reply:
Okay I've updated the comments, and as an added bonus corrected the spelling from Paypload -> Payload.

If you'd like me to, I can squash the commits.

Reply:
A sorry for not being clear enough: I didn't just mean the comments - also the code is not completely correct. In case there is a payload, the `BaseLayer.Contents` shouldn't hold all the data, but just the part that belongs to this layer (so everything except the Payload).

Reply:
Aha.   Now I understand.   I had the wrong assumption that even though there was another logical layer (SDP) since there was no actual decoder layer, that the base layer should include everything.

I've reworked the patch to properly separate LayerContents and Payload.